pkg/mapper/service.go

```diff
@@ -72,13 +94,13 @@ func (fs *Service) MapFn(stream mappb.Map_MapFnServer) error {
 	// Read requests from the stream and process them
 outer:
 	for {
-		select {
-		case <-groupCtx.Done():
+		req, err := recvWithContext(groupCtx, stream)
+		if errors.Is(err, context.Canceled) {
```

Comment:
I see different behaviour between map and map batchmap above. batchmap is returning the err while map is breaking the outer loop without returning the err. Is this expected? same for map stream.